lib/pipeline-importing.js

```diff
@@ -1,26 +1,32 @@
-import { uuid } from 'mu';
-import { STATUS_BUSY,
-         STATUS_FAILED,
-         STATUS_SUCCESS,
-         TARGET_GRAPH,
-       } from '../constants';
+import { uuid, sparqlEscapeUri } from 'mu';
+import { querySudo as query } from '@lblod/mu-auth-sudo';
 
-import {  updateTaskStatus, appendTaskError } from './task';
-import { writeTriplesToGraph,
-         appendTaskResultGraph,
+import {
+  STATUS_BUSY,
+  STATUS_FAILED,
+  STATUS_SUCCESS,
+  TARGET_GRAPH,
+} from '../constants';
+
+import { updateTaskStatus, appendTaskError } from './task';
+import {
+  writeTriplesToGraph,
+  appendTaskResultGraph,
   getTriples
-       } from './graph';
-import { triplesToNT } from './utils';
+} from './graph';
+import { triplesToNT, filterAsync, processPart } from './utils';
 const BATCH_SIZE = process.env.BATCH_SIZE || 100;
 
-export async function run( task ){
+export async function run(task) {
   try {
     await updateTaskStatus(task, STATUS_BUSY);
 
     //Some detour in moving data from one graph to another is needed, cause:
     // 1. mu-auth doesn't support COPY statements
     // 2. a simple insert/where is generates too big statements for the database. Batching is not legal sparql
-    const triples = await getTriples(task);
+    let triples = await getTriples(task);
+    triples = await filterAsync(triples, (triple) => checkIfExist(TARGET_GRAPH, triple), true)
```

Comment:
There is a problem it will shoot many concurrent  calls to the DB, and I am afraid it just might get overloaded. 
I came across another way of checking existence (Thanks to @madnificent ) which might be potentially very fast
```
SELECT DISTINCT ?s ?p ?o WHERE {
  VALUES (?s ?p ?o) {
  (<http://baz> <http://bar> <http://boom>)
  }
  ?s ?p ?o
}
```
Here you can bundle your calls, can we try this? (This will be needed as long as we don't have this fixed at mu-auth)

Reply:
I get a 500, I'll test without specifying the graph and will let you know:
```
harvesting-sameas_1                                      |   
harvesting-sameas_1                                      | Headers set on SPARQL client: {"requestDefaults":{"headers":{"mu-auth-sudo":"true"}}}
harvesting-sameas_1                                      | 
harvesting-sameas_1                                      |     SELECT DISTINCT ?s ?p ?o WHERE {
harvesting-sameas_1                                      |       graph <http://mu.semte.ch/graphs/public> {
harvesting-sameas_1                                      |         VALUES (?s ?p ?o) {
harvesting-sameas_1                                      |           (<http://data.lblod.info/id/d1924d20-5e55-11ec-928c-8d5a31e1d6bf> <http://www.w3.org/ns/prov#generated> <http://data.lblod.info/id/d1cd0c30-5e55-11ec-928c-8d5a31e1d6bf>)
harvesting-sameas_1                                      |         }
harvesting-sameas_1                                      |         ?s ?p ?o
harvesting-sameas_1                                      |       }
harvesting-sameas_1                                      |     }
harvesting-sameas_1                                      |   
harvesting-sameas_1                                      | Headers set on SPARQL client: {"requestDefaults":{"headers":{"mu-auth-sudo":"true"}}}
harvesting-sameas_1                                      | Error: HTTP Error: 500 Internal Server Error
harvesting-sameas_1                                      |     at /usr/src/app/app/node_modules/sparql-client-2/lib/client.js:60:29
harvesting-sameas_1                                      |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
harvesting-sameas_1                                      |     at checkIfExist (/usr/src/app/app/lib/pipeline-importing.js:59:16)
harvesting-sameas_1                                      |     at async Promise.all (index 74)
harvesting-sameas_1                                      |     at run (/usr/src/app/app/lib/pipeline-importing.js:29:15)
harvesting-sameas_1                                      |     at /usr/src/app/app/app.js:38:9 {
harvesting-sameas_1                                      |   httpStatus: 500
harvesting-sameas_1                                      | }

Reply:
same when not specifying the graph; 

the ask query  in the current PR is working fine for the current link i'm testing but I don't know if it shoots many concurrent calls to the db:

<img width="541" alt="Capture d’écran 2021-12-16 à 11 12 29" src="https://user-images.githubusercontent.com/3816305/146352286-7b6bab23-a2e6-469a-b8cd-0d64f895f630.png">

```
    ask {
      graph ${sparqlEscapeUri(graph)} {
        ${subject} ${predicate} ${object}.
      }
    }